doc/api/punycode.md

```diff
@@ -1,6 +1,11 @@
 # punycode
 
-    Stability: 2 - Stable
+    Stability: 0 - Deprecated
+
+**The version of the punycode module bundled in Node.js is being deprecated**.
+In a future major version of Node.js this module will be removed. Users
```

Comment:
@mathiasbynens ... the best bet for your users would be to begin recommending transitioning off the bundled version to the installable version starting in v7, even tho it's just a "soft deprecation". Ideally those users would be migrated already long before we do the hard deprecation (where warnings and errors start being reported). In fact, the faster users transition off the bundled version, the faster we'll be able to do that hard deprecation.